apps/system/js/app_text_selection_dialog.js

```diff
@@ -16,13 +19,20 @@
     this._enabled = false;
     this._shortcutTimeout = null;
     this._injected = false;
-    this._hasCutOrCopied = false;
     this._isCommandSendable = false;
     this._transitionState = 'closed';
     this.textualmenuDetail = null;
     this.bindOnObservePrefChanged = this.onObservePrefChanged.bind(this);
     SettingsListener.observe('copypaste.enabled', true,
                              this.bindOnObservePrefChanged);
+
+    Object.defineProperty(this, '_hasCutOrCopied', {
+      get: function() { return !!_cutOrCopiedTimeoutId; },
```

Comment:
The one and only state the GlobalStates instance is `hasCutOrCopied`, which can be found by looking at the timeout ID. Therefore, timeoutId should be kept in the GlobalStates instance.